be/src/exec/tablet_scanner.cpp

```diff
@@ -49,12 +49,17 @@ Status TabletScanner::init(RuntimeState* runtime_state, const TabletScannerParam
     RETURN_IF_ERROR(Expr::clone_if_not_exists(runtime_state, &_pool, *params.conjunct_ctxs, &_conjunct_ctxs));
     RETURN_IF_ERROR(_get_tablet(params.scan_range));
 
-    // if column_desc come from fe, reset tablet schema
-    if (_parent->_olap_scan_node.__isset.columns_desc && !_parent->_olap_scan_node.columns_desc.empty() &&
-        _parent->_olap_scan_node.columns_desc[0].col_unique_id >= 0) {
-        _tablet_schema = TabletSchema::copy(*_tablet->tablet_schema(), _parent->_olap_scan_node.columns_desc);
-    } else {
-        _tablet_schema = _tablet->tablet_schema();
+    if (_parent->_olap_scan_node.__isset.schema_id) {
+        _tablet_schema = GlobalTabletSchemaMap::Instance()->get(_parent->_olap_scan_node.schema_id);
+    }
+    if (_tablet_schema == nullptr) {
+        // if column_desc come from fe, reset tablet schema
+        if (_parent->_olap_scan_node.__isset.columns_desc && !_parent->_olap_scan_node.columns_desc.empty() &&
+            _parent->_olap_scan_node.columns_desc[0].col_unique_id >= 0) {
+            _tablet_schema = TabletSchema::copy(*_tablet->tablet_schema(), _parent->_olap_scan_node.columns_desc);
+        } else {
+            _tablet_schema = _tablet->tablet_schema();
+        }
```

Comment:
[be/src/exec/pipeline/scan/olap_chunk_source.cpp](https://github.com/StarRocks/starrocks/blob/main/be/src/exec/pipeline/scan/olap_chunk_source.cpp#L412)